synapse/handlers/federation.py

```diff
@@ -1498,6 +1522,36 @@ async def _sync_partial_state_room(
                 allow_rejected=True,
             )
             for event in events:
-                await self._federation_event_handler.update_state_for_partial_state_event(
-                    destination, event
-                )
+                for attempt in range(len(destinations)):
```

Comment:
given that we never expect to hit the end of this for loop (aiui?) it would probably be clearer to replace it with `while True:` and manually incrementing `attempt`.

Reply:
`for attempt in itertools.count():`?